### ISSUE TITLE
Help page: Indicate multi-input options

### ISSUE THREAD

**Issue by Lestropie** [MRtrix3/mrtrix3#132]
When generating the help page, use some sort of indication that multiple argument input files / multiple uses of an option is permitted (i.e. over and above the help text for that particular argument/option). Likewise generate a similar indication for options that have the required flag set.

**Comment by jdtournier** [comment]
In reference to #336 (show description of arguments to options on help page), this used to be how it was done in older versions (just check out the help page for any command in the 0.2.x version). In MRtrix3, these are explicitly ignored - it's not a bug (although this clearly could do with being documented). The reason for this is that in pretty much all cases, this introduces pointless redundancy in the help page, making the help page needlessly longer and more verbose, and forcing developers to write essentially the same information twice. For example:

~~~
+ Option ("-lmax", "specify maximum harmonic degree to use in reconstruction")
+   Argument ("value", "the maximum harmonic degree to use").type_integer (2, 8, 1e6);
~~~

Clearly, the description of the `value` argument is completely redundant, given that it's explained perfectly clearly in the Option description. This type of repetition happens pretty much every time - at least it happened enough that I decided it was better to ignore these fields for Options and just make sure the Option itself documented its arguments clearly. In practice, I found it much more natural to describe all these arguments there anyway, since they naturally make more sense in the context of the option they relate to.

What I might do is update the documentation to clearly indicate that the description field of Arguments to Option is ignored, these should be documented in the Option description. Could also add an `assert()` to catch these cases and alert the developer earlier...?

Note this is also an example where setting a default value is actually ill-defined (I'm talking about #334 here): in many cases, the default `lmax` used would depend on the input data. So it's another good reason to ditch these default values...
